src/main/java/org/broadinstitute/hellbender/tools/examples/metrics/single/ExampleSingleMetrics.java

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.hellbender.tools.examples.metrics.single;
+
+import htsjdk.samtools.metrics.MetricBase;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.Serializable;
+
+public class ExampleSingleMetrics extends MetricBase implements Serializable {
+    private static final long serialVersionUID = 1;
+
+    public int numReads = 0;
+
+    //adds stats for the read
+    public ExampleSingleMetrics addRead(final GATKRead read) {
```

Comment:
Yes, we should have something similar to what we have in for the Spark layer, in some future PR.